### PR TITLE
Adding allow return outside function option to babel options

### PR DESCRIPTION
**Pull request by helloIAmPau**
Hi all,
due to some internal needs of my company, I added the babylon "allowReturnOutsideFunction" option to the babel base option object.

This way you can switch the flag by using the configuration object  passed as second argument to the `transform` method.

I hope you enjoy the PR.
cheers

### CODE REVIEW

**Comment by hzoo** [comment]
If we want to do this, I think we probably want to support passing the other babylon options in as well

**Comment by helloIAmPau** [comment]
You are right. If you want I can update the PR by adding the other babylon options.

**Comment by helloIAmPau** [comment]
I updated the pr by adding the other babylon options.
Let me know if everything fits.

Bye

**Comment by hzoo** [comment]
This works but I wonder if its better to just pass options down to babylon/babel-generator so you don't have to manually do this for each new option that is added (@loganfsmyth mentioned this)

**Comment by helloIAmPau** [comment]
For me is OK, but in order to do that I have to prevent the option checking.
Look at the file `packages/babel-core/src/transformation/file/options/config.js`, here the options are declared explicitly.

If you want, I can investigate further, but I don't think that skipping the checking is the good solution.

File: packages/babel-core/src/transformation/file/options/config.js [hunk 1]
```diff
@@ -39,6 +39,30 @@ module.exports = {
...
+  },
+
+  strictMode: {
+    description: "TODO",
+    type: "string",
+    default: null
+  },
+
+  allowSuperOutsideMethod: {
+    description: "When enabled, import/export statements are not constrained to appearing at the top of the program.",
```
Comment (hzoo) [inline packages/babel-core/src/transformation/file/options/config.js]: This should be switched with `allowImportExportEverywhere`? https://github.com/babel/babel/pull/3503/files

**Comment by snuggs** [comment]
ETA/Status on this? 👍 if reify.js sent you here. ;-) /cc @hzoo @ helloIAmPau

**Comment by hzoo** [comment]
I think we should be able to incorporate this kind of change into https://github.com/babel/babel/pull/3503

**Comment by danez** [comment]
Incorporate in itself? 😂

**Comment by hzoo** [comment]
oops.. #3561 sigh I should take a break

**Comment by snuggs** [comment]
LMAO @danez I was like this... 
dog-chasing-tail

**Comment by hzoo** [comment]
Will be doing this in https://github.com/babel/babel/pull/3561 under parserOpts

**Pull request closed without merging**